### ISSUE TITLE
Feature request: Error handling event type or agent?

### ISSUE THREAD

**Issue by virtadpt** [huginn/huginn#2498]
I realize this should probably go in the agent suggestions thread, but I think this goes a bit beyond a regular agent into the realm of good error handling.

Let's take a situation where an RSS Agent gets an error message from the server it's contacting because the site no longer supports RSS:

```
Failed to fetch http://datalossdb.org/blotter_posts.rss: #<Faraday::Response:0x0000000ae6e088 @on_complete_callbacks=[], @env=#<Faraday::Env @method=:get @body="<!DOCTYPE html>...
```

What if, when an agent throws an exception like this, it threw a special event containing a capsule form of the error, which could then be detected either by a Trigger Agent or a special Error Handling Agent, and at the very least made it possible to alert the user of Huginn that something went awry and should be attended to (aside from looking at the Logs tab of every agent)?

**Comment by brianpetro** [comment]
Good idea. Sometimes I manually implement catching errors and outputting
them in the event payload. Then using a Trigger agent to detect and handle
events with an error.

**Comment by brianpetro** [comment]
Two additional checkboxes for each agent configuration. One that catches
all errors and emits as an error-event when checked. And another that
toggles acceptance of error-events from the source agent.

**Comment by virtadpt** [comment]
How do you manually implement error catching and handling?  Do you have an exported scenario anywhere?

I really like the checkbox option.

**Comment by jasongwq** [comment]
#1957

**Comment by mdrichardson** [comment]
I'd like to bump this is there's a couple issues on this that haven't gone anywhere:

1. https://github.com/huginn/huginn/issues/1957
2. https://github.com/huginn/huginn/issues/2394

I think @brianpetro's idea is pretty good, but I think a simpler solution might be something like:

1. For each agent, add another input box below "Receivers" called "Error Receivers"
2. Have all agents catch all raised errors and send them as "Error Events" to the "Error Receivers" (might have to manually add `rescue`'s to `check()`?)

I tried doing something like this with a fork of the repo, but ran into `ActiveRecord::HasManyThroughSourceAssociationNotFoundError:` errors when adding `has_many :error_receivers, :through => :links_as_source, :class_name => "Agent", :inverse_of => :sources` (along with many other changes to try this) . Not knowing Ruby or Huginn, I didn't want to spend the time tracking that one down. If anybody has some pointers, I'm all ears and could maybe work on this sporadically.
